### PR TITLE
Feature/feature notification

### PR DESCRIPTION
**Pull request by danicm47**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  12
         

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java  -2
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPeriodicNotification.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePeriodicNotification.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PeriodEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPeriodicNotification.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePeriodicNotification.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036&bid=17740261)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerAdapter.java [hunk 1]
```diff
@@ -242,6 +247,16 @@ public void deletePetWashFrequency(User user, Pet pet, DateTime dateTime) {
 
     }
 
+    @Override
+    public void registerNewPeriodicNotification(User user, Pet pet, Event event, int period) {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerAdapter.java:267]: You should implement the access to the library

**Review by AlbertPG — CHANGES_REQUESTED** [review]
You should implement the access to the services

File: app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerService.java [hunk 2]
```diff
@@ -110,4 +112,21 @@
...
     void deletePetWashFrequency(User user, Pet pet, DateTime dateTime);
+
+    /**
+     * Add a periodic event to pet.
+     * @param user The user that wants to add a periodic event
+     * @param pet The pet which the user wants to add a periodic event
+     * @param event The period event
+     * @param period The periodicity of the period event
+     */
+    void registerNewPeriodicNotification(User user, Pet pet, Event event, int period) throws ParseException, ExecutionException, InterruptedException;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerService.java:123]: Codacy Issue found: [Line is longer than 120 characters (found 150).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePeriodicNotification.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PeriodEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePeriodicNotification.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPeriodicNotification.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPeriodicNotification.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036&bid=17740261)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerService.java [hunk 3]
```diff
@@ -110,4 +112,21 @@
...
+
+    /**
+     * Delete a periodic event of pet.
+     * @param user The user that wants to delete a periodic event
+     * @param pet The pet which the user wants to delete a periodic event
+     * @param event The period event that user wants to delete
+     */
+    void deletePeriodicEvent(User user, Pet pet, Event event) throws ParseException, ExecutionException, InterruptedException;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerService.java:131]: Codacy Issue found: [Line is longer than 120 characters (found 126).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPeriodicNotification.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PeriodEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPeriodicNotification.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePeriodicNotification.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePeriodicNotification.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036&bid=17740261)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPeriodicNotification.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPeriodicNotification.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PeriodEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePeriodicNotification.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePeriodicNotification.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036&bid=17740261)

**Review by AlbertPG — APPROVED** [review]
Okey, I approve it

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePeriodicNotification.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PeriodEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePeriodicNotification.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPeriodicNotification.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPeriodicNotification.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036&bid=17740261)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPeriodicNotification.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePeriodicNotification.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPeriodicNotification.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePeriodicNotification.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PeriodEvent.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5411036&bid=17740261)

**Pull request merged**